Performance docs for Android

marandaneto opened the pull request:
Add Performance docs for Android
Also fixes #2849

marandaneto committed:
Performance docs for Android

marandaneto commented on src/docs/product/performance/getting-started.mdx:
```diff
@@ -17,12 +17,17 @@ Customers on legacy plans must add transaction events to their subscription in o
   - [React](/platforms/javascript/guides/react/#monitor-performance)
   - [Vue](/platforms/javascript/guides/vue/#monitor-performance)
 
+- Mobile
```
@bruno-garcia not sure about a Mobile section here but under Java would be kinda weird if we now have a Kotlin language, wdyt?

bruno-garcia replied:
Make sense to me to categorize like you did. We could rename Java to [Java/Kotlin/JVM] ?

marandaneto committed:
fix

marandaneto committed:
fix

bruno-garcia commented on src/docs/product/performance/getting-started.mdx:
```diff
@@ -17,12 +17,17 @@ Customers on legacy plans must add transaction events to their subscription in o
 
 ## Supported SDKs
 
+- [Java](/platforms/java/performance/)
```
```suggestion
- [Java/Kotlin/JVM](/platforms/java/performance/)
```

bruno-garcia commented on src/includes/performance/enable-tracing/android.mdx:
````diff
@@ -0,0 +1,38 @@
+</application>
+```
+
+Or, if you are manually instrumenting Sentry:
````
Opting in to tracing with the sampling decision isn't really related to manual vs auto instrumentation (we can start capturing transactions automatically if we are able to if the users set a sampling decision in):

```suggestion
Or, if you are initializing the SDK programatically:
```

marandaneto replied:
this was the wording defined by the docs team about "auto init mode" vs "initializing the SDK by yourself".
I'd stick to it or we'd need to change everywhere, otherwise, only this page would differ.

it's not really related to tracing, but if you have the auto init mode or not

bruno-garcia replied:
@PeloWriter maybe has an opinion here

bruno-garcia commented on src/includes/performance/enable-tracing/android.mdx:
````diff
@@ -0,0 +1,38 @@
+
+Or, if you are manually instrumenting Sentry:
+
+```java {tabTitle:Java}
````
What about having Kotlin first?

Maybe better as a single PR across all of docs, right?

marandaneto replied:
I'd say so, also would need checking if docs somehow would order alphabetically or not

bruno-garcia commented on src/includes/performance/traces-sample-rate/android.mdx:
````diff
@@ -0,0 +1,26 @@
+</application>
+```
+
+Or, if you are manually instrumenting Sentry:
````
```suggestion
Or, if you are initializing the SDK programatically:
```

bruno-garcia commented on src/includes/performance/traces-sample-rate/android.mdx:
```diff
@@ -0,0 +1,26 @@
+
+SentryAndroid.init(this,
+  options -> {
+    ...
```
```suggestion
```

marandaneto replied:
this is in all the performance Java/Kotlin examples, if we decide to remove it, I'd do a single PR.

bruno-garcia commented on src/includes/performance/traces-sampler-as-sampler/android.mdx:
```diff
@@ -0,0 +1,62 @@
+      context -> {
+        CustomSamplingContext ctx = context.getCustomSamplingContext();
+        if (ctx != null) {
+          if ("/payment".equals(ctx.get("url"))) {
+            // These are important - take a big sample
+            return 0.5;
+          } else if ("/search".equals(ctx.get("url"))) {
+            // Search is less important and happen much more frequently - only take 1%
+            return 0.01;
+          } else if ("/health".equals(ctx.get("url"))) {
```
Maybe something more "mobile"?

```suggestion
          if ("LoginActivity".equals(ctx.get("ActivityName"))) {
            // These are important - take a big sample
            return 0.5;
          } else if ("SearchActivity".equals(ctx.get("ActivityName"))) {
            // Search is less important and happen much more frequently - only take 1%
            return 0.01;
          } else if ("HealthCheckActivity".equals(ctx.get("ActivityName"))) {
```

marandaneto replied:
good catch, makes sense

marandaneto committed:
Merge branch 'master' into feat/android-performance

marandaneto committed:
fix

marandaneto committed:
move to includes

marandaneto committed:
fix

marandaneto committed:
fix code snippets

marandaneto committed:
fix snippets

marandaneto commented:
@maciejwalkowiak I've changed a few scripts to be compatible with the new code like using interfaces instead of the implementation class, also nullability checks, added imports, and returning Doubles instead of Ints otherwise it'd never compile.
Please check, thanks

maciejwalkowiak reviewed (approved):
Looks good to me.

marandaneto commented on src/includes/java/performance/getting-started/java.mdx:
```diff
@@ -0,0 +1,168 @@
+
+The next example contains the implementation of the hypothetical `processItem` function called from the code snippet in the previous section. Our SDK can determine if there is currently an open transaction and add all newly created spans as child operations to that transaction. Keep in mind that each individual span also needs to be manually finished; otherwise, spans will not show up in the tran ...
+
+In cases where you want to attach Spans to an already ongoing Transaction you can use `Sentry#getSpan`. This method will return a `SentryTransaction` in case there is a running Transaction or a `Span` in case there is already a running Span, otherwise it returns `null`.
```
@maciejwalkowiak should we still keep `SentryTransaction` and `Span` or just `ISpan`?

marandaneto commented on src/includes/java/performance/getting-started/java.mdx:
```diff
@@ -0,0 +1,168 @@
+
+### Retrieving a Transaction
+
+In cases where you want to attach Spans to an already ongoing Transaction you can use `Sentry#getSpan`. This method will return a `SentryTransaction` in case there is a running Transaction or a `Span` in case there is already a running Span, otherwise it returns `null`.
```
same as https://github.com/getsentry/sentry-docs/pull/2851/files#r557241698

marandaneto committed:
fix

bruno-garcia reviewed (approved):
🚀
